ipn/backend.go

```diff
@@ -185,8 +185,23 @@ type Options struct {
 	//    state and use/update that.
 	//  - StateKey!="" && Prefs!=nil: like the previous case, but do
 	//    an initial overwrite of backend state with Prefs.
+	//
+	// NOTE(apenwarr): This means that this Prefs field does not do
+	// what you probably think it does. It will overwrite your encryption
+	// keys. Do not use unless you know what you're doing.
 	StateKey StateKey
 	Prefs    *Prefs
+	// UpdatePrefs, if provided, overrides Options.Prefs *and* the Prefs
```

Comment:
So you're saying that if UpdatePrefs is non-nil, then Options.Pref is ignored?

But `startIsNoopLocked` looks at Prefs (whether it's non-nil). So this comment isn't quite complete.

Reply:
Also, `startIsNoopLocked` now has an outdated comment:

```go
        // Options has 4 fields; check all of them:                                                                                                                           
        //   * FrontendLogID                                                                                                                                                  
        //   * StateKey                                                                                                                                                       
        //   * Prefs                                                                                                                                                          
        //   * AuthKey                                                                                                                                                        
```